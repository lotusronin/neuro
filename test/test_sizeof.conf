conf = {
'src' : 'test_sizeof.nro',
'expected' : 'test_sizeof.expected'
}
