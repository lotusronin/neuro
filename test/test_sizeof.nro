import stdio;

fn main() : u32 {
    sizeu8 := sizeof(u8);
    sizeu16 := sizeof(u16);
    sizeu32 := sizeof(u32);
    sizeu64 := sizeof(u64);
    sizes8 := sizeof(s8);
    sizes16 := sizeof(s16);
    sizes32 := sizeof(s32);
    sizes64 := sizeof(s64);
    sizef32 := sizeof(f32);
    sizef64 := sizeof(f64);
    
    stdio::puts("Sizeof u8");
    stdio::printi(sizeu8);
    stdio::putchar('\n');
    stdio::puts("Sizeof u16");
    stdio::printi(sizeu16);
    stdio::putchar('\n');
    stdio::puts("Sizeof u32");
    stdio::printi(sizeu32);
    stdio::putchar('\n');
    stdio::puts("Sizeof u64");
    stdio::printi(sizeu64);
    stdio::putchar('\n');
    
    stdio::puts("Sizeof s8");
    stdio::printi(sizes8);
    stdio::putchar('\n');
    stdio::puts("Sizeof s16");
    stdio::printi(sizes16);
    stdio::putchar('\n');
    stdio::puts("Sizeof s32");
    stdio::printi(sizes32);
    stdio::putchar('\n');
    stdio::puts("Sizeof s64");
    stdio::printi(sizes64);
    stdio::putchar('\n');
    
    stdio::puts("Sizeof f32");
    stdio::printi(sizef32);
    stdio::putchar('\n');
    stdio::puts("Sizeof f64");
    stdio::printi(sizef64);
    stdio::putchar('\n');
    return 0;
}
