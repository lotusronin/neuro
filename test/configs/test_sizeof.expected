Sizeof u8
1
Sizeof u16
2
Sizeof u32
4
Sizeof u64
8
Sizeof s8
1
Sizeof s16
2
Sizeof s32
4
Sizeof s64
8
Sizeof f32
4
Sizeof f64
8
